src/components/config-tabs/denominators/DenominatorTableItem.js

```diff
@@ -0,0 +1,134 @@
+import {
+    DELETE_DENOMINATOR,
+    UPDATE_DENOMINATOR,
+    useConfigurationsDispatch,
+    useDataItemNames,
+} from '../../../utils/index.js'
+import { ConfirmationModal } from '../ConfirmationModal.js'
+import { EditDenominatorModal } from './EditDenominatorModal.js'
+
+/** Manages the "update form" modal and datastore mutation */
+const EditDenominatorButton = ({ denominator }) => {
+    const [editModalOpen, setEditModalOpen] = useState(false)
+    const dispatch = useConfigurationsDispatch()
+
+    const openModal = useCallback(() => setEditModalOpen(true), [])
+    const closeModal = useCallback(() => setEditModalOpen(false), [])
+
+    const updateDenominator = useCallback(
+        ({ newDenominatorData }) => {
+            console.log('newDenominatorData', newDenominatorData)
+            dispatch({
+                type: UPDATE_DENOMINATOR,
+                payload: {
+                    code: denominator.code,
+                    updatedDenominatorData: newDenominatorData,
+                },
+            })
+        },
+        [dispatch, denominator.code]
+    )
+
+    // not all fields are needed for form initial values
+    const denominatorDataForForm = useMemo(
+        () => ({
+            code: denominator.code,
+            type: denominator.type,
+            level: denominator.lowLevel,
+            dataID: denominator.dataID,
+        }),
+        [denominator]
+    )
+
+    return (
+        <>
+            <Button small onClick={openModal}>
+                Edit
+            </Button>
+            {editModalOpen && (
+                <EditDenominatorModal
+                    denominatorDataToEdit={denominatorDataForForm}
+                    onSave={updateDenominator}
+                    onClose={closeModal}
+                />
+            )}
+        </>
+    )
+}
+EditDenominatorButton.propTypes = {
+    denominator: PropTypes.object,
+}
+
+const DeleteDenominatorButton = ({ denominator }) => {
+    const [confirmationModalOpen, setConfirmationModalOpen] = useState(false)
+
+    const dispatch = useConfigurationsDispatch()
+
+    const openModal = useCallback(() => setConfirmationModalOpen(true), [])
+    const closeModal = useCallback(() => setConfirmationModalOpen(false), [])
+
+    const deleteDenominator = useCallback(
+        () =>
+            dispatch({
+                type: DELETE_DENOMINATOR,
+                payload: { code: denominator.code },
+            }),
+        [dispatch, denominator.code]
+    )
+
+    return (
+        <>
+            <Button
+                small
+                destructive
+                onClick={() => {
+                    openModal()
+                }}
+            >
+                Delete
+            </Button>
+            {confirmationModalOpen && (
+                <ConfirmationModal
+                    title="Delete denominator"
+                    text={`Are you sure you want to delete ${denominator.name}?`}
```

Comment:
name is not a property on the denominators you are creating, so you need to use `dataItemNames.get(denominator.dataID)`

Reply:
This is right because even though i just updated the code to collect name from user inputs, we're not showing the user the name on the table.